web/client/plugins/GeoStory.jsx

```diff
@@ -28,19 +35,49 @@ import MediaContentToolbar from './geostory/MediaContentToolbar';
 const GeoStory = ({
     story,
     mode = Modes.VIEW,
+    onAdd,
+    messages,
+    fontFamilies = [],
+    storyFonts = [],
+    onUpdateSetting = () => {},
+    onBasicError = () => {},
     ...props
-}) => (<BorderLayout
-    className="ms-geostory"
-    columns={[<MapEditor {...props} mode={mode} />]}>
-    <Story
-        {...story}
-        {...props} // add actions
-        mode={mode}
-        mediaViewer={MediaViewer}
-        contentToolbar={MediaContentToolbar}
-    />
-</BorderLayout>
-);
+}) => {
+    const localize = useCallback((id) => getMessageById(messages, id), [messages]);
+    const addFunc = (path, position, element) => onAdd(path, position, element, localize);
+
+    useEffect(() => {
+        onUpdateSetting("fontFamilies", fontFamilies);
+    }, [ fontFamilies ]);
```

Comment:
- [ ] If fontFamilies is not configured in localConfig this go in infinite loop because ad each render it will recieve a new empty array (see fontFamilies in prop list). Probably this could be solved using GeoStory.defaultProps instead. Steps:
  - remove fontFamilies from localConfig
  - use debug=true
  - login as admin
  - go to homepage
  - select 'Demo Observatories' story
  - infinite in console `GEOSTORY:UPDATE_SETTING` in console